### PR TITLE
Add the ability to undo optimization

### PR DESCRIPTION
**Pull request by Atala**
undo-optim

### CODE REVIEW

**Review by vladimir-8 — APPROVED** [review]

**Comment by vladimir-8** [comment]
Just a note, `Optimise` icon has a nice tooltip, but other icons don't

Screenshot 2024-07-19 at 14 37 47

File: src/Service/RouteOptimizer.php [hunk 1]
```diff
@@ -94,7 +94,10 @@ function($desc) use ($iriConverter, $items) {
             })->toArray()
         );
 
-        return $res;
+        return [
+            "solution" => $res,
+            "unassigned_count" => $data["summary"]["unassigned"]
+        ];
```
Comment (alexsegura) [inline src/Service/RouteOptimizer.php:100]: Maybe we could return an object here.

File: src/Action/TaskList/Optimize.php [hunk 2]
```diff
@@ -3,27 +3,36 @@
...
-        return $data;
+        return new JsonResponse([
+            "solution" => $this->taskListNormalizer->normalize($data, 'jsonld', [
+                'resource_class' => TaskList::class,
+                'operation_type' => 'item',
+                'item_operation_name' => 'get',
+                'groups' => ['task_list']
+            ]),
+            "unassigned_count" => $optim["unassigned_count"]
+        ]);
```
Comment (alexsegura) [inline src/Action/TaskList/Optimize.php:36]: We should return an object here, and let the serializer do its job.

File: src/Action/TaskList/Optimize.php [hunk 3]
```diff
@@ -3,27 +3,36 @@
...
 
-        return $data;
+        return new JsonResponse([
+            "solution" => $this->taskListNormalizer->normalize($data, 'jsonld', [
+                'resource_class' => TaskList::class,
+                'operation_type' => 'item',
+                'item_operation_name' => 'get',
+                'groups' => ['task_list']
+            ]),
+            "unassigned_count" => $optim["unassigned_count"]
```
Comment (alexsegura) [inline src/Action/TaskList/Optimize.php:35]: In other parts of the code, JSON responses are generally camel case. 

Should be `unassignedCount`

**Comment by Atala** [comment]
Merged manually

**Pull request closed without merging**